### PR TITLE
ChatOps Plugin Support

### PR DESCRIPTION
**Pull request by dhollinger**
Build plugin for Chatops tools to replace the `notify_` methods within the helpers.

Still a work in progress. Initial support being built for Slack.

Next PR will support Rocket.chat to bring the app in-line with the `puppet-r10k` script.

### CODE REVIEW

File: lib/helpers/tasks.rb [hunk 1]
```diff
@@ -65,59 +65,46 @@ def generate_types(environment)
     notify_slack(status_message) if slack?
```
Comment (ekohl) [inline lib/helpers/tasks.rb]: This (and line 61) should be `notification()`

Comment (dhollinger) [reply lib/helpers/tasks.rb]: Nice catch! I'll take care of that.

File: lib/plugins/chatops/slack.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        message = {
+          author: 'r10k for Puppet',
+          title: "r10k deployment of Puppet environment #{target}"
+        }
+
+        case @message[:status_code]
+        when 202
+          message.merge!(
+            color: 'good',
+            text: "Successfully deployed #{target}",
```
Comment (ekohl) [inline lib/plugins/chatops/slack.rb]: 202 is technically just forked and can still fail. Should this be `Deployment started`?

File: spec/unit/plugins/webhook_chatops_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        options: {
+          icon_emoji: ':ocean:'
+        }
+      }
+
+      allow(PuppetWebhook::Chatops).to receive(:notify).and_return(notification_data)
+      expect(PuppetWebhook::Chatops.notify).to eq(notification_data)
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (dhollinger) [reply spec/unit/plugins/webhook_chatops_spec.rb]: @codacy-bot there is a new line there

**Pull request merged**